### PR TITLE
Add ICR to Digital Carbon

### PR DESCRIPTION
**Pull request by cujowolf**
Schema changes:

Enums: Add ICR to both `Registry` and `BridgeProtocol`

`tokenId` field added to the `ProvenanceRecord` and `Holding` entities

CarbonCredit entity additions:
- `tokenAddress` - Ethereum token address for the credit
- `tokenId` - Token ID for the credit (if applicable)
- `isExAnte` - Boolean flag for if the credit is an exAnte credit.
- `exPostTokenId` - Linked exPost token ID if this credit is an exAnte credit.

`Retire` entity additions:
- `hash` field representing the transaction hash of the retirement.

### CODE REVIEW

File: polygon-digital-carbon/src/KlimaAggregator.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { MossRetired } from '../generated/RetireMossCarbon/RetireMossCarbon'
 import { ToucanRetired } from '../generated/RetireToucanCarbon/RetireToucanCarbon'
 import { C3Retired } from '../generated/RetireC3Carbon/RetireC3Carbon'
-import { CarbonRetired } from '../generated/KlimaInfinity/KlimaInfinity'
+import { CarbonRetired, CarbonRetired1 as CarbonRetiredTokenId } from '../generated/KlimaInfinity/KlimaInfinity'
```
Comment (0xAurelius) [inline polygon-digital-carbon/src/KlimaAggregator.ts:4]: What's with `CarbonRetired1`? I don't see that function defined in the ABI for KlimaInfinity...

File: polygon-digital-carbon/src/TransferHandler.ts [hunk 2]
```diff
@@ -1,62 +1,196 @@
...
+  hash: Bytes,
+  logIndex: i32,
+  timestamp: BigInt,
+  blockNumber: BigInt
+): void {
+  let tokenVar = ''
+  if (tokenId !== null) {
+    tokenVar = tokenId.toString()
+  }
+  log.debug(
```
Comment (0xAurelius) [inline polygon-digital-carbon/src/TransferHandler.ts:145]: Did you mean to leave this debug logger in here?

**Review by 0xAurelius — APPROVED** [review]
Overall LGTM, just a couple minor comments

File: polygon-digital-carbon/src/KlimaAggregator.ts (hunk 1 above)
Comment (cujowolf) [reply polygon-digital-carbon/src/KlimaAggregator.ts:4]: This is the new event emission that shows up on the Aggregator due to emitting a token ID along with the address for a carbon project for ICR retirements. https://github.com/KlimaDAO/klimadao-solidity/blob/main/src/infinity/libraries/Bridges/LibICRCarbon.sol#L15

**Pull request merged**